src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -3310,6 +3343,173 @@ .maxstack  3
 ");
         }
 
+        [Fact]
+        public void ArraysSupport()
```

Comment:
>ArraysSupport [](start = 20, length = 13)

Consider testing:
- fixed size buffers
- field initializers

Reply:
> fixed size buffers

These can only be bool, byte, short, int, long, char, sbyte, ushort, uint, ulong, float or double types. I'll test to make sure that function pointers are disallowed.